File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        //int Size = 0;
        ListNode* FastNode = head;
        ListNode* SlowNode = head;
      while(FastNode != nullptr && FastNode->next != nullptr)
      {
        FastNode = FastNode->next->next;
        SlowNode = SlowNode->next;
      }
      
      return SlowNode;
      
    }
};

File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode MiddleNode(ListNode head) {
          ListNode FastNode = head;
        ListNode SlowNode = head;
      while(FastNode != null && FastNode.next != null)
      {
        FastNode = FastNode.next.next;
        SlowNode = SlowNode.next;
      }
      
      return SlowNode;
      
    }  
}